core/src/main/java/io/substrait/relation/physical/NestedLoopJoin.java

```diff
@@ -0,0 +1,80 @@
+package io.substrait.relation.physical;
+
+import io.substrait.expression.Expression;
+import io.substrait.proto.NestedLoopJoinRel;
+import io.substrait.relation.BiRel;
+import io.substrait.relation.HasExtension;
+import io.substrait.relation.RelVisitor;
+import io.substrait.type.Type;
+import io.substrait.type.TypeCreator;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class NestedLoopJoin extends BiRel implements HasExtension {
+
+  public abstract Optional<Expression> getCondition();
```

Comment:
What do you think about making this
```java
public abstract Expression getCondition();
```

The spec indicates that
```proto
  // optional, defaults to true (a cartesian join)
  Expression expression = 4;
```

but we can be stricter within the POJO layer and require it. You're already inserting a boolean literal condition if the protobuf doesn't contain one https://github.com/substrait-io/substrait-java/pull/188/files#r1372502462

Reply:
I can go ahead and change the spec, since I just added it recently.

Reply:
Hmm, it's worth noting that `JoinRel` uses ```public abstract Optional<Expression> getCondition();```, which is required. Maybe it is best to implement them similarly?

Reply:
I made the change in the latest commit if it is the preferred choice!

Reply:
The JoinRel in the protobuf spec doesn't have any information about whether condition is required or not:
https://github.com/substrait-io/substrait/blob/b3071bc9cd77cf916568641c83056a285f8123be/proto/substrait/algebra.proto#L156-L160

I'm guessing that's why it's Optional currently. I think it would make sense to make the JoinRel condition be required as well. That can be a separate change though.